src/test/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriterRoundtripTest.java

```diff
@@ -56,124 +61,160 @@ public static Stream<File> data() throws Exception {
-        if (null == tiffImageMetadata) {
+
+        if (tiffImageMetadata == null) {
             throw new TestSkippedException();
         }
+
         return tiffImageMetadata;
     }
 
     private TiffOutputSet getTiffOutputSet(final TiffImageMetadata sourceTiffImageMetadata) throws ImagingException {
+
         final TiffOutputSet tiffOutputSet = sourceTiffImageMetadata.getOutputSet();
+
         if (tiffOutputSet == null) {
             throw new TestSkippedException();
         }
+
         return tiffOutputSet;
     }
 
     @AfterEach
     void tearDown() {
+
         if (duplicateFile != null && duplicateFile.exists()) {
             duplicateFile.delete();
             duplicateFile.deleteOnExit();
         }
     }
 
-    @ParameterizedTest
-    @MethodSource("data")
-    public void updateExifMetadataLossless_copyWithoutChanges_TiffImageMetadataIsIdentical(final File sourceFile) throws Exception {
-        /*
-         * Load EXIF data from source file, skipping over any test images without any EXIF data
-         */
-        final JpegImageMetadata sourceJpegImageMetadata = getJpegImageMetadata(sourceFile);
-        final TiffImageMetadata sourceTiffImageMetadata = getTiffImageMetadata(sourceJpegImageMetadata);
-        final TiffOutputSet sourceTiffOutputSet = getTiffOutputSet(sourceTiffImageMetadata);
-
-        /*
-         * Copy the TiffOutputSet to a duplicate TiffOutputSet
-         */
-        final TiffOutputSet duplicateTiffOutputSet = duplicateTiffOutputSet(sourceTiffOutputSet);
-
-        /*
-         * Copy the file to a duplicate file, using updateExifMetadataLossless and the duplicate TiffOutputSet
-         */
-        copyToDuplicateFile(sourceFile, duplicateTiffOutputSet);
-
-        /*
-         * Load EXIF data from duplicate file
-         */
-        final JpegImageMetadata duplicateJpegImageMetadata = getJpegImageMetadata(duplicateFile);
-        final TiffImageMetadata duplicateTiffImageMetadata = getTiffImageMetadata(duplicateJpegImageMetadata);
-
-        /*
-         * Compare the source TiffImageMetadata to the one loaded from the duplicate file. This fails!
-         */
-        final List<? extends ImageMetadata.ImageMetadataItem> imageMetadataItems = sourceTiffImageMetadata.getItems();
-        final List<? extends ImageMetadata.ImageMetadataItem> imageMetadataItems1 = duplicateTiffImageMetadata.getItems();
-        assertEquals(imageMetadataItems.size(), imageMetadataItems1.size(), "The TiffImageMetadata have different numbers of imageMetadataItems.");
-
-        for (int i = 0; i < imageMetadataItems.size(); i++) {
-            final ImageMetadata.ImageMetadataItem imageMetadataItem = imageMetadataItems.get(i);
-            final ImageMetadata.ImageMetadataItem imageMetadataItem1 = imageMetadataItems1.get(i);
-            assertEquals(imageMetadataItem.toString(), imageMetadataItem1.toString(), "ImageMetadataItem toString mismatch.");
-        }
-    }
-
```

Comment:
This toString() comparison test is useless.

The TiffWriter does change the order of fields in an attempt to optimize for space. This results in fields being reordered and unnecessary offset fields to be dropped.

You only can compare the TiffOutputSet.

I assume the author did this to compare values. This is why I introduced the value comparison API to TiffOutputField.